src/helpers.ts

```diff
@@ -48,6 +50,10 @@ CHART_LOG_INTERVALS.set('1w', 7 * 24 * 60 * 60)
 
 export const pricePrecision = BigInt.fromI32(2).pow(96)
 
+export function normalizeDailyTimestamp(timestamp: BigInt): BigInt {
+  return timestamp.minus(timestamp.mod(BigInt.fromI32(86400)))
```

Comment:
If the timestamp of the snapshot is A, that snapshot is expected to hold data before A.
So I think the normalization should be as below:
```suggestion
  return timestamp.add(BigInt.fromI32(86399)).div(BigInt.fromI32(86400))
```

Reply:
I think this expression is incorrect. If the current timestamp is 1741740540, the result will be 20160. I want to get 1741737600 instead @JhChoy